src/modules/FairProjectConfig.cmake

```
@@ -0,0 +1,103 @@
+################################################################################
+# Copyright (C) 2018-2021 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  #
+#                                                                              #
+#              This software is distributed under the terms of the             #
+#              GNU Lesser General Public Licence (LGPL) version 3,             #
+#                  copied verbatim in the file "LICENSE"                       #
+################################################################################
+
+if(CMAKE_VERSION VERSION_LESS 3.12)
+	message(FATAL_ERROR "Module FairProjectConfig requires CMake 3.12 or later!")
+endif()
+
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+*****************
+FairProjectConfig
+*****************
+
+.. versionadded:: 1.0.0
+
+Requires CMake 3.12 or later.
+
+#]=======================================================================]
+
+
+#[=======================================================================[.rst:
+---------
+
+``fair_get_git_version()``
+==========================
+
+.. code-block:: cmake
+
+   fair_get_git_version([OUTVAR_PREFIX prefix]
+                        [TAGS_ONLY] [REQUIRED])
+
+The function checks git attributes of the current tree.
+If it can't detect anything useful (becausee Git is not
+installed, or because there is no git repository), it will
+not touch any output variables at all. 
+
+``OUTVAR_PREFIX``
+    This option sets the prefix for created
+    variables. It defaults to ``"PROJECT"``.
+
+``TAGS_ONLY``
+    This option tells Git to search for tags and use them
+    as base for the returned information.
+    If this option is not given, Git can also return a
+    (shortened) commit hash.
+    (Current implementation detail: not passing this option
+    will pass ``--always`` to ``git describe``.)
+
+``REQUIRED``
+    If no suitable information can be detected (for example
+    ``TAGS_ONLY`` is given, but no tags exist, or no git
+    repository exists) then a fatal error is raised.
+
+Output variables:
+
+:variable:`<prefix>_GIT_VERSION`
+    A version string that is created from the output of
+    ``git describe``.
+
+#]=======================================================================]
+find_package(Git QUIET)
+function(fair_get_git_version)
+  cmake_parse_arguments(PARSE_ARGV 0 ARGS
+                        "REQUIRED;TAGS_ONLY" "OUTVAR_PREFIX" "")
+
+  if(NOT ARGS_OUTVAR_PREFIX)
+    set(ARGS_OUTVAR_PREFIX PROJECT)
+  endif()
+
+  if(NOT ARGS_TAGS_ONLY)
+    set(git_always "--always")
+  endif()
+
+  if(GIT_FOUND AND EXISTS ${CMAKE_SOURCE_DIR}/.git)
```

```suggestion
  if(Git_FOUND AND EXISTS ${CMAKE_SOURCE_DIR}/.git)
```

The find module also sets `GIT_FOUND` but `Git_FOUND` is the more modern convention.

---

the existance check for `.git` is fragile, consider using `git rev-parse --is-inside-work-tree` or similar (not sure how recent the `--is-inside-work-tree` flag is, also see e.g. https://remarkablemark.org/blog/2020/06/05/check-git-repository/)

---

What about dropping the existence check completely and letting `git describe` handle it?

```console
CMake Error at src/modules/FairProjectConfig.cmake:91 (message):
  git describe failed (return value 128):

  fatal: not a git repository (or any parent up to mount point /)

  Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```